### ISSUE TITLE
Unexpected error for api.query.proxy.proxies

### ISSUE THREAD

**Issue by mkchungs** [polkadot-js/api#4948]
I'm getting an unexpected error when trying to parse proxy:proxies record for this particular account: 1KvKReVmUiTc2LW2a4qyHsaJJ9eE9LRsywZkMk5hyBeyHgw 
https://polkaholic.io/account/1KvKReVmUiTc2LW2a4qyHsaJJ9eE9LRsywZkMk5hyBeyHgw

- Steps to Reproduce the Error (on [polkadot.js](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.polkadot.io#/js) - endpoint: wss://rpc.polkadot.io)
```
let testKey = '1KvKReVmUiTc2LW2a4qyHsaJJ9eE9LRsywZkMk5hyBeyHgw'
let proxyAddrs = []
let keys = await api.query.proxy.proxies.keys()
for (const key of keys) {
      proxyAddrs.push(key.args.map((k) => k.toHuman())[0])
}
console.log(`total proxyAddrs = ${proxyAddrs.length}`)
console.log(`testKey ${testKey} found at proxy.proxies.keys() idx=${proxyAddrs.indexOf(testKey)}`)

let proxiesVal = await api.query.proxy.proxies(testKey) //0x0e6de68b13b82479fbe988ab9ecb16bad446b67b993cdd9198cd41c7c6259c49
console.log(proxiesVal)
```
- Error msg:
```
total proxyAddrs = 510

testKey 1KvKReVmUiTc2LW2a4qyHsaJJ9eE9LRsywZkMk5hyBeyHgw found at proxy.proxies.keys() idx=136

Error: Unable to decode storage proxy.proxies: entry 0:: createType((Vec<PalletProxyProxyDefinition>,u128)):: decodeU8a: failed at 0x0898c8a3d01da9877b7b30877c717ae8…: Vec<PalletProxyProxyDefinition>:: decodeU8a: failed at 0x1da9877b7b30877c717ae8f2a7e726ce…: {"delegate":"AccountId32","proxyType":"PolkadotRuntimeProxyType","delay":"u32"}:: decodeU8a: failed at 0xbd2e0000000000000000000000… on proxyType: {"_enum":{"Any":0,"NonTransfer":1,"Governance":2,"Staking":3,"__Unused4":4,"IdentityJudgement":5,"CancelProxy":6,"Auction":7}}:: Unable to create Enum via index 189, in Any, NonTransfer, Governance, Staking, __Unused4, IdentityJudgement, CancelProxy, Auction
    at Zs._newType (https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:76042)
    at Zs._formatStorageSetEntry (https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:75563)
    at https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:75315
    at r.reduce (<anonymous>)
    at Zs._formatStorageSet (https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:75285)
    at https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:74998
    at Array.map (<anonymous>)
    at Zs._formatOutput (https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:74941)
    at Zs._formatResult (https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:72595)
    at i (https://polkadot.js.org/apps/polkadot.01.fa5dbabf.js:1:73062)
```

Can you provide some insights on this error? is there a new proxyType list?

**Comment by jacogr** [comment]
The definition for ProxyType actually comes from the metadata itself, so the types are correct.

In this case, you actually found an unmigrated storage item in a previous format. So in the runtime, the current data is in the form `{"delegate":"AccountId32","proxyType":"PolkadotRuntimeProxyType","delay":"u32"}` however this specific storage entry does not have the (newer) `delay` parameter at all.

Looking at the storage data itself -

```
0x
// 8 << 2 == 2 (number of items)
08
// 14TKt6bUNjKJdfYqVDNFBqzDAwmJ7WaQwfUmxmizJHHrr1Gs
98c8a3d01da9877b7b30877c717ae8f2a7e726cefa176c5dfcdcebc9b6a12223
// Any
00
// 14TKt6bUNjKJdfYqVDNFBqzDAwmJ7WaQwfUmxmizJHHrr1Gs
98c8a3d01da9877b7b30877c717ae8f2a7e726cefa176c5dfcdcebc9b6a12223
// SudoBalances
04
// 20.0740 DOT
005109bd2e0000000000000000000000
```

Basically it is still in the old `(AccountId, ProxyType)` form and cannot be decoded with the types specified by the runtime metadata at all. (Since the delay parameter is completely missing in the on-chain data)

(If you query the state at an ancient block, e.g. `300599` (hash `0x39f1f69a0faa148dbc0fdf3206029d474cb2f3c5d6aa59f777f06c2605c8cbd5`) which is just after both proxies were set, you will see the older results, and at this point it can decode it.

When the new proxy storage was introduced in the runtime, old entries were not migrated. This is one of them.

**Comment by mkchungs** [comment]
thanks for the insight!
